src/content/docs/en/guides/backend/turso.mdx

```diff
@@ -0,0 +1,144 @@
+- An Access Token
+
+If you don't have an existing database and schema with Turso, create one and connect to the shell using the Turso CLI:
+
+```bash
+turso db create my-astro-db
+turso db shell my-astro-db
+```
+
+Then add the following:
+
+```sql
+CREATE TABLE posts (
+    id INT AUTO_INCREMENT PRIMARY KEY,
+    title TEXT,
+    slug TEXT UNIQUE,
+    content TEXT
+);
+```
+
+Make sure to populate the `posts` table with some content. You'll need it to make the most out of the examples below.
+
+### Configure environment variables
+
+You can obtain your database URL using the following command:
+
+```bash
+turso db show my-astro-db --url
+```
+
+Then we'll need to create an auth token for the database:
+
+```bash
+turso db tokens create my-astro-db
+```
+
+Now create the file `.env` and add the output from both commands above:
+
+```ini title=".env"
+TURSO_DATABASE_URL=libsql://...
+TURSO_AUTH_TOKEN=
+```
+
+:::caution
+Make sure you **do not** expose the `TURSO_DATABASE_URL` and `TURSO_AUTH_TOKEN` on the client with the `PUBLIC_` prefix when creating [environment variables](/en/guides/environment-variables/).
+:::
+
+## Install LibSQL Client
+
+To connect Astro to Turso we will need to install the `@libsql/client`:
+
+<PackageManagerTabs>
+  <Fragment slot="npm">
+    ```shell
+    npm install @libsql/client
+    ```
+  </Fragment>
+  <Fragment slot="pnpm">
+    ```shell
+    pnpm install @libsql/client
+    ```
+  </Fragment>
+  <Fragment slot="yarn">
+    ```shell
+    yarn add @libsql/client
+    ```
+  </Fragment>
+</PackageManagerTabs>
+
+## Initialize a new client
+
+Create a file `turso.ts` in the `src` folder and invoke `createClient`, passing it `TURSO_DATABASE_URL` and `TURSO_AUTH_TOKEN`:
+
+```ts
+import { createClient } from "@libsql/client/web";
+
+export const turso = createClient({
+  url: import.meta.env.TURSO_DATABASE_URL,
+  authToken: import.meta.env.TURSO_AUTH_TOKEN,
+});
+```
+
+## Create a query
+
+Inside `src/pages/index.astro`, import `turso` and execute a SQL query to fetch all `posts` from the table:
+
+```astro title="src/pages/index.astro"
+---
+import {turso} from '../turso'
+
+const posts = await turso.execute('SELECT * FROM posts')
+---
+
+<ul>
+  {posts.map((post) => (
+    <li>{post.title}</li>
+  ))}
+</ul>
+```
+
+## Create a query with arguments
+
+There are cases where you'll need to pass values to your SQL query for things like a `slug`, `id`, or pagination. Thankfully, the `execute` method also accepts an object that safely passes your variables to the SQL statement.
+
+Create a file `src/pages/posts/[slug].astro`, with the following:
+
+```astro title="src/pages/posts/[slug].astro"
+---
+import {turso} from '../../turso'
+
+const {slug} = Astro.params
+
+const post = await turso.execute({
+  sql: 'SELECT * FROM posts WHERE slug = ?',
+  args: [slug!]
+})
+---
+
+<pre>{JSON.stringify(post, null, 2)}</pre>
+```
+
+That's it! Turso is now set up and configured to read content from the database with Astro.
```

Comment:
This example needs to be restructured.

Use the previous structure as an example: First, say what you can DO by passing an argument in generalities. Then, use a format like "The following example shows..." retrieving? querying for? and explain what that query is fetching and how it is used below.

Separate what this code is supposed to do from **how it works**. When we explain code samples, we tell them first, in words, what the code actually does. Then, when they inspect the code, they can simply focus on **how they use your tool to do that**. In this case, there's no guidance to the reader about what you're trying to do, what info is being fetched or intended to be displayed. So they have to ALSO figure out that from just the code sample, in addition to what you're actually trying to show them, which is **how to use Turso to do that**.

And, no need for that last line at all.

Reply:
@sarah11918 I pushed a few changes which I'll try to tune even further later today or tomorrow.